src/NClient/NClient.Standalone/ClientProxy/Validation/Transport/StubTransportRequestBuilderProvider.cs

```diff
@@ -1,13 +1,20 @@
 ﻿using NClient.Providers;
 using NClient.Providers.Transport;
+using NClient.Providers.Transport.Common;
 
 namespace NClient.Standalone.ClientProxy.Validation.Transport
 {
-    internal class StubTransportRequestBuilderProvider : ITransportRequestBuilderProvider<IRequest, IResponse>
+    internal class StubTransportRequestBuilderProvider<TRequest, TResponse> : ITransportRequestBuilderProvider<IRequest, IResponse>
     {
         public ITransportRequestBuilder<IRequest, IResponse> Create(IToolset toolset)
         {
-            return new StubTransportRequestBuilder();
+            return new StubTransportRequestBuilder<TRequest, TResponse>();
+        }
+
+        public ITransportRequestBuilder<IRequest, IResponse> Create(IToolset toolset, IPipelineCanceller pipelineCanceller)
+        {
+            pipelineCanceller.Cancel();
```

Comment:
It seems that this is not required here, since there is such a check in `ClientInterceptor`:
```C#
if (_pipelineCanceler.IsCancellationRequested)
{
    _toolset.Logger?.LogDebug("Validation halting pipeline execution");
    return default; //return a non-null so the pipeline can exit gracefully
}
```

Reply:
Hey @smolchanovsky, that `.Cancel()` in the `TransportRequestBuilderProvider` is there to cancel the pipeline execution, from [this comment](https://github.com/nclient/NClient/issues/218#issuecomment-1230321602) in the issue (but we later decided to use` CancelationToken` instead of exception but I kept the logic the same). If I remove this `.Cancel()` then the pipeline won't terminate at the correct place?
> you can terminate the pipeline execution when the request has already been created. In this case, I suggest to create internal exception, for example ValidationSuccessfulFinishedException (heh, oxymoron) and always throw it in StubTransportRequestBuilderProvider

Reply:
@reaver330 Hi! I really apologize for not answering. Bad things are happening in my country, so I don't have the opportunity to do opensource right now. I hope I will be able to answer within a week